src/Hamiltonians/excitations.jl

```diff
@@ -139,6 +135,27 @@ function momentum_transfer_diagonal(map::BoseOccupiedModeMap)
     end
     return float(onproduct)
 end
+
+"""
+    extended_momentum_transfer_diagonal(map, M)
+
+The diagonal part of nearest neighbour [`momentum_transfer_excitation`](@ref).
+"""
+
+function extended_momentum_transfer_diagonal(map::OccupiedModeMap,M::Int)
+    onproduct = 0
+    step = (2*π)/M
```

Comment:
```suggestion
function extended_momentum_transfer_diagonal(map::OccupiedModeMap, step)
    onproduct = 0
```
I would suggest to pre-calculate `step` in the calling function, where it can be done by the compiler. In this function the number of modes is run-time data and the (expensive) floating-point division would have to be done at run-time.